gtd_generative_proof_machine/inc/generative_restriction.h

```diff
@@ -0,0 +1,60 @@
+#ifndef GENERATIVE_RESTRICTION_H
+#define GENERATIVE_RESTRICTION_H
+
+#include "common.h"
+#include "physical_graph.h"
+#include "generative_proof_machine.h"
+
+typedef struct GenerativeRestriction GenerativeRestriction;
+typedef struct intNode intNode;
+typedef struct PathNode PathNode;
+
+typedef struct PathNode
+{
+    int *path;
+    int *pathBool;
+    int numElems;
+    PathNode *next;
+} PathNode;
+
+typedef struct RestrictionResult
+{
+    int modified;
+    int contradictionFound;
```

Comment:
for boolean use bool from stdbool or uint8_t (usually equivalent in C), no need to use 32 bits here